diff --git a/include/tool/dl_tool.hpp b/include/tool/dl_tool.hpp
index e5490e0..5deddbd 100644
--- a/include/tool/dl_tool.hpp
+++ b/include/tool/dl_tool.hpp
@@ -388,9 +388,9 @@ namespace dl
             void print()
             {
 #if DL_LOG_LATENCY_UNIT
-                printf("latency: %15u cycle\n", this->get_average_period());
+                printf("latency: %15lu cycle\n", this->get_average_period());
 #else
-                printf("latency: %15u us\n", this->get_average_period());
+                printf("latency: %15lu us\n", this->get_average_period());
 #endif
             }
 
@@ -402,9 +402,9 @@ namespace dl
             void print(const char *message)
             {
 #if DL_LOG_LATENCY_UNIT
-                printf("%s: %15u cycle\n", message, this->get_average_period());
+                printf("%s: %15lu cycle\n", message, this->get_average_period());
 #else
-                printf("%s: %15u us\n", message, this->get_average_period());
+                printf("%s: %15lu us\n", message, this->get_average_period());
 #endif
             }
 
@@ -417,9 +417,9 @@ namespace dl
             void print(const char *prefix, const char *key)
             {
 #if DL_LOG_LATENCY_UNIT
-                printf("%s::%s: %u cycle\n", prefix, key, this->get_average_period());
+                printf("%s::%s: %lu cycle\n", prefix, key, this->get_average_period());
 #else
-                printf("%s::%s: %u us\n", prefix, key, this->get_average_period());
+                printf("%s::%s: %lu us\n", prefix, key, this->get_average_period());
 #endif
             }
         };
